docs/development.md

```diff
@@ -35,3 +35,39 @@ Since this is very unlikely and most times an abs path is not strictly necessary
 "Should" functions can be used anywhere,
 but care should be taken not to overuse them when a fallback value is not sufficient.
 "Must" variants should be favored in testing code, since they will more obviously fail tests.
+
+## Cross-Platform Scripting
+
+We are writing bash scripts with the intention of running them in a POSIX environment.
+However, differences between operating systems, particularly between Linux and MacOS,
+can still affect a script's behavior. 
+This section highlights key considerations to ensure your scripts run consistently across platforms.
+
+### Regular Expressions
+Differences between how Linux and MacOS handle regular expressions can cause inconsistent outcomes when writing scripts.
+Be aware of the following:
+
+- **BRE (Basic Regular Expressions)**
+  - **Usage**: Default in many tools.
+  - **Consideration**: Limited feature set.
+  - **[More Info](https://en.wikibooks.org/wiki/Regular_Expressions/Basic_Regular_Expressions)**
+
+- **ERE (Extended Regular Expressions)**
+  - **Usage**: Enabled with `-E` flag in tools like `grep` and `sed`.
```

Comment:
I would soften this statement up a bit, since you can't say it is `-E` in every (or even most) tools.
Maybe something like: "Often enabled with the `-E` flag (like in `grep` and `sed`).